### ISSUE TITLE
fix: iOS crash after patch (unknown crash point)

### ISSUE THREAD

**Issue by moweiran** [shorebirdtech/shorebird#1681]
I am a subscribe customer. I need help. My project release by shorebird cmd below.
```
shorebird release ios-alpha -f --no-codesign --flavor prod -- --dart-define=env=prod
shorebird patch ios-alpha --flavor prod -- --dart-define=env=prod
```
Because my project has development and production environment. So the project scheme has two option one is dev other is prod, And also the project name is "AstroStar Dev" for dev, And "AstroStar" for prod.
But when i use patch. The below information show me at terminal. I continue and upload patch successfully. 
When i install the new version app, Then download the patch and relaunch, Crash report!!!!!.
```
[WARN] Your app contains native changes, which cannot be applied with a patch.
    Changed files:
        Products/Applications/AstroStar Dev.app/AstroStar Dev

If you don't know why you're seeing this error, visit our troublshooting page at https://docs.shorebird.dev/troubleshooting#unexpected-native-changes
```

**Steps To Reproduce**

1. shorebird release ios-alpha -f --no-codesign --flavor prod -- --dart-define=env=prod
2. shorebird patch ios-alpha --flavor prod -- --dart-define=env=prod
3. Installed App
4. Relaunch App
5. Crash.

```
lzy@LZYdeMac-mini astrostar-app % shorebird doctor
Shorebird 0.24.0 • git@github.com:shorebirdtech/shorebird.git
Flutter 3.16.7 • revision 48bdf4e4bf91a040b187986c01a790727809023c
Engine • revision 2b094773ca4b350d5a6f51504ddac68fb0964832

✗ Shorebird is up-to-date (1.2s)
  [!] A new version of shorebird is available! Run `shorebird upgrade` to upgrade.
✓ Flutter install is correct (0.5s)
✓ AndroidManifest.xml files contain INTERNET permission (46ms)
✓ Has access to storage.googleapis.com (0.6s)
1 issue detected.
```
Here is the crash report.

[live.astrostar.dev_issue_289a3c868840634daf068a36cb5111f9_crash_session_e649671c66c3495a81b01b9a3fe57b0e_DNE_0_v2_stacktrace.txt](https://github.com/shorebirdtech/shorebird/files/14119292/live.astrostar.dev_issue_289a3c868840634daf068a36cb5111f9_crash_session_e649671c66c3495a81b01b9a3fe57b0e_DNE_0_v2_stacktrace.txt)

**Comment by eseidel** [comment]
Thank you for the crash stack.  We will look right away.  It looks like 0.24.0 still had a crasher slip through.

**Comment by eseidel** [comment]
The stack trace ends in 0xffffffffe which suggests that its somehow running Clang code on the Dart stack. Since the Dart stack commonly has 0xffffffffe in it to signal exiting the simulator.

**Comment by eseidel** [comment]
There is also the same address printed a bunch of times, as though perhaps this is a recursion?
0x112b8a188 shows up 11 times in the stack.

**Comment by eseidel** [comment]
Actually, new theory.  EndTransitionDartToCpuIfNeeded was removed from our Flutter several releases ago.

**Comment by felangel** [comment]
We took a closer look and it appears as though your iOS release was using an older version of Flutter (3.16.4). There are many known iOS crashers in older versions of Shorebird's Flutter Engine and we highly recommend upgrading to the latest Flutter version and creating a new release + patch. Let us know if you're still experiencing crashes after doing so and we're happy to take a closer look, thanks!

```
# Make sure to use the latest version of Shorebird CLI (and the Shorebird Flutter Engine)
shorebird upgrade

shorebird flutter versions use 3.16.7
```

**Comment by eseidel** [comment]
For what it's worth, this report has caused us to fix at least two issues in our handling of Flutter versions, so hopefully you're the last person who will ever see this crash.  We don't currently plan to back-port our new iOS engine fixes to these older Flutter revisions, so we'll be releasing an update to the CLI later today which will prevent releasing with these specific known-bad versions of Flutter.  Thank you again for the report!  Please let us know if upgrading to 3.16.7 does not fix.

**Comment by eseidel** [comment]
I'm going to close this on the assumption that `shorebird flutter versions use 3.16.7` should resolve the issue.  *please* let us know if that does not.  We're happy to jump on a call to help if necessary.

**Comment by moweiran** [comment]
> For what it's worth, this report has caused us to fix at least two issues in our handling of Flutter versions, so hopefully you're the last person who will ever see this crash. We don't currently plan to back-port our new iOS engine fixes to these older Flutter revisions, so we'll be releasing an update to the CLI later today which will prevent releasing with these specific known-bad versions of Flutter. Thank you again for the report! Please let us know if upgrading to 3.16.7 does not fix.

Yes, I have upgrading to 3.16.7, but it's not working.

**Comment by eseidel** [comment]
We would like to help. We also just released 3.16.9. Maybe I was wrong
about 3.16.7 working.

https://calendly.com/eseidel/shorebird-dem
<https://calendly.com/eseidel/shorebird-demo?back=1&month=2024-02>o shows
my calendar if you’d like us to hop on a call with you to debug. We’re also
reachable on discord.gg/shorebird

**Comment by moweiran** [comment]
Hi, @eseidel 
I have try again after shorebird cli upgrade to lastest version 0.25.0, and the flutter version use 3.16.9.
This time we unused patch. When i open the app, and switch some page. It's crash again when i installed by testflight. 
[live.astrostar.dev_issue_75ef8ae30ba8d40c959b60434daa2147_crash_session_7f175fa10af84c00913a358b5328a1de_DNE_0_v2_stacktrace.txt](https://github.com/shorebirdtech/shorebird/files/14138847/live.astrostar.dev_issue_75ef8ae30ba8d40c959b60434daa2147_crash_session_7f175fa10af84c00913a358b5328a1de_DNE_0_v2_stacktrace.txt)

### I release app by normal steps and upload to testflight.  It's worked and no crash.

```
lzy@LZYdeMac-mini astrostar-app % shorebird release ios-alpha -f --no-codesign --flavor dev -- --dart-define=env=dev
Building for device with codesigning disabled. You will have to manually codesign before deploying to device.
[WARN] shorebird preview will not work for releases created with "--no-codesign". However, you can still preview your app by signing the generated .xcarchive in Xcode.
✓ Fetching apps (0.8s)
✓ Building release (133.6s)
✓ Fetching releases (0.9s)

🚀 Ready to create a new release!

📱 App: astrostar (dev) (d5055e67-c13a-4453-8d84-e51d9a345fa9)
🍧 Flavor: dev
📦 Release Version: 2.10.0+83
🕹️  Platform: ios
🐦 Flutter Version: 3.16.9 (84abdc0bad)

✓ Creating release (0.5s)
✓ Creating artifacts (176.5s)
✓ Updating release status (0.6s)

✅ Published Release 2.10.0+83!

Your next step is to submit the archive at build/ios/archive/AstroStar Dev.xcarchive to the App Store using Xcode.

You can open the archive in Xcode by running:
    open build/ios/archive/AstroStar Dev.xcarchive

Make sure to uncheck "Manage Version and Build Number", or else shorebird will not work.
```

**Comment by eseidel** [comment]
Thank you again for you patience and repeated assistance.  We will work to symbolicate that crash correctly and get back to you in a few hours, thanks.

**Comment by eseidel** [comment]
We attempted to symbolicate locally based on the provide crash stack and were unsuccessful.  However we've updated our infrastructure to upload .dSYM files which can be provided to crashlytics to have it include symbols in its output.

@bryanoltman is working on publishing instructions on how to do so as we speak.  Thank you again for your help.

**Comment by bryanoltman** [comment]
The dsym for our version of Flutter 3.16.9 can be downloaded from https://storage.googleapis.com/download.shorebird.dev/flutter_infra_release/flutter/dcbbf162662ed6c2539db27d4322d719808cb5f9/ios-release/Flutter.dSYM.zip. If you can upload that to Crashlytics from the dSYMs tab (see image below) and report back once the crash logs have been updated, that would be incredibly helpful

![Image](https://github.com/shorebirdtech/shorebird/assets/581764/79da7f27-f26c-4949-9945-3f735a6d3c91)

**Comment by eseidel** [comment]
OK, we've found a crasher which may be causing this.  Should have a fix out within a day or so.  If you have a chance to upload the .dsym to crashlytics and post a symbolicated crash log that would still be helpful.  Thank you!

**Comment by eseidel** [comment]
We've added a new guide on how to get the corresponding .dsym files to upload them to your service for more detailed reports: https://docs.shorebird.dev/guides/crash-reporting

We have a release coming out today which hopefully will fix this crash. 🤞  Thank you again for your patience.

**Comment by eseidel** [comment]
We believe this crash was fixed by Shorebird 0.25.2 last week, but we've now also released Shorebird 0.26.0 which fixes other known crashers in iOS.  If you're still seeing any issues we would *love* to help.  But barring any new data, I'm going to presume this is resolved?  Please let us know if you have any trouble.
